pkg/registry/core/pod/etcd/eviction.go

```diff
@@ -116,6 +116,10 @@ func (r *EvictionREST) Create(ctx api.Context, obj runtime.Object) (runtime.Obje
 }
 
 func (r *EvictionREST) checkAndDecrement(namespace string, podName string, pdb policy.PodDisruptionBudget) (ok bool, err error) {
+	if pdb.Status.ObservedGeneration == nil || *pdb.Status.ObservedGeneration != pdb.Generation {
```

Comment:
If i understand correctly each object has a version that is incremented every update. It is possible to allow update only if the internal etcd version equals to the version in the update. With that no 2 concurrent updates would be able to use the same budget at the same time. The time the first one gets in the version is incremented and the second one fails.